### PR TITLE
[LIBCLOUD-649] Add new methods for LinodeNodeDriver

### PR DESCRIPTION
**Pull request by wirkijowski**
Unit tests included
Jira ticket: https://issues.apache.org/jira/browse/LIBCLOUD-649

### CODE REVIEW

**Comment by Kami** [comment]
Thanks.

I quickly glanced over the changes and they look good to me.

I'm just wondering, if we can make "ex_create_volume" and "ex_list_volume" methods some how comply with the standard API. Looking at the Linode API docs, LinodeId argument is required so sadly this might not be possible.

File: libcloud/compute/drivers/linode.py [hunk 1]
```diff
@@ -480,6 +484,137 @@ def linode_set_datacenter(self, dc):
...
+
+    def ex_list_volumes(self, node, disk_id=None):
+        """
+        List existing disk volumes for for given Linode.
+
+        :keyword    node: Node to list disk volumes for. (required)
+        :type       node: :class:`Node`
+
+        :keyword    disk_id: Id for specific disk volume. (optional)
+        :keyword     disk_id: ``int``
```
Comment (Kami) [inline libcloud/compute/drivers/linode.py]: Should be `:type disk_id:`

**Comment by wirkijowski** [comment]
In order to comply with standart libcloud API, there could be create_volume() method. There are two ways it could be done. 
The first one is create_volume() could return StorageVolume instance which would be created for real after attach_volume(). But this would need that programmer should remeber that they operate on nonexistent volume until it's attached. 
The second way is to use create_volume() with additional ex_ params. But this params need to be required.

ex_list_volumes() could be rewritten to list_volumes() with optional ex_ params, definig node or disk_id.

**Comment by Kami** [comment]
Sorry for the delay.

It sounds like we should just stay with extension methods for now since none of the approaches makes it fully compliant with the base API, in fact, it might just make it more confusing to the end user (it's named as the base API, but it doesn't behave as such).

**Comment by Kami** [comment]
Squashed the commits and merged changes into trunk. Thanks.

**Comment by wirkijowski** [comment]
Thanks:)

**Pull request closed without merging**